src/System.Linq.Expressions/tests/Array/NewArrayBoundsTests.cs

```diff
@@ -0,0 +1,251 @@
+                    yield return new object[] { typeof(char), size, sizeType, (char)0 };
+                    yield return new object[] { typeof(decimal), size, sizeType, (decimal)0 };
+                    yield return new object[] { typeof(double), size, sizeType, (double)0 };
+                    yield return new object[] { typeof(float), size, sizeType, (float)0 };
+                    yield return new object[] { typeof(int), size, sizeType, 0 };
+                    yield return new object[] { typeof(long), size, sizeType, (long)0 };
+                    yield return new object[] { typeof(S), size, sizeType, new S() };
+                    yield return new object[] { typeof(sbyte), size, sizeType, (sbyte)0 };
+                    yield return new object[] { typeof(Sc), size, sizeType, new Sc() };
+                    yield return new object[] { typeof(Scs), size, sizeType, new Scs() };
+                    yield return new object[] { typeof(short), size, sizeType, (short)0 };
+                    yield return new object[] { typeof(Sp), size, sizeType, new Sp() };
+                    yield return new object[] { typeof(Ss), size, sizeType, new Ss() };
+                    yield return new object[] { typeof(uint), size, sizeType, (uint)0 };
+                    yield return new object[] { typeof(ulong), size, sizeType, (ulong)0 };
+                    yield return new object[] { typeof(ushort), size, sizeType, (ushort)0 };
+
+                    // Object
+                    yield return new object[] { typeof(C), size, sizeType, null };
+                    yield return new object[] { typeof(D), size, sizeType, null };
+                    yield return new object[] { typeof(Delegate), size, sizeType, null };
+                    yield return new object[] { typeof(E), size, sizeType, (E)0 };
+                    yield return new object[] { typeof(El), size, sizeType, (El)0 };
+                    yield return new object[] { typeof(Func<object>), size, sizeType, null };
+                    yield return new object[] { typeof(I), size, sizeType, null };
+                    yield return new object[] { typeof(IEquatable<C>), size, sizeType, null };
+                    yield return new object[] { typeof(IEquatable<D>), size, sizeType, null };
+                    yield return new object[] { typeof(object), size, sizeType, null };
+                    yield return new object[] { typeof(string), size, sizeType, null };
+                }
+            }
+        }
+
+        [Theory]
+        [PerCompilationType(nameof(TestData))]
+        public static void NewArrayBounds(Type arrayType, object size, Type sizeType, object defaultValue, bool useInterpreter)
+        {
+            Expression newArrayExpression = Expression.NewArrayBounds(arrayType, Expression.Constant(size, sizeType));
+            Expression <Func<Array>> e = Expression.Lambda<Func<Array>>(newArrayExpression);
```

Comment:
I do not think Array vs T[] is material here since we are testing NewArrayBounds factory and it takes the array type as ```Type arrayType``` - factory is not generic.
Array vs T[]  only affect casting of the result before validation.